src/LabelStudio.js

```diff
@@ -62,20 +64,68 @@ export class LabelStudio {
     this.store = store;
     window.Htx = this.store;
 
-    render((
-      <App
-        store={this.store}
-        panels={registerPanels(this.options.panels) ?? []}
-      />
-    ), rootElement);
+    const isRendered = false;
+
+    const renderApp = () => {
+      if (isRendered) {
+        clearRenderedApp();
+      }
+      render((
+        <App
+          store={this.store}
+          panels={registerPanels(this.options.panels) ?? []}
+        />
+      ), rootElement);
+    };
+
+    const clearRenderedApp = () => {
+      const childNodes = [...rootElement.childNodes];
+      // cleanDomAfterReact needs this key to be sure that cleaning affects only current react subtree
+      const reactKey = findReactKey(childNodes[0]);
 
-    const destructor = () => {
       unmountComponentAtNode(rootElement);
+      /*
+        Unmounting doesn't help with clearing React's fibers
+        but removing the manually helps
+        @see https://github.com/facebook/react/pull/20290 (similar problem)
+        That's maybe not relevant in version 18
+       */
+      cleanDomAfterReact(childNodes, reactKey);
+      cleanDomAfterReact([rootElement], reactKey);
+    };
+
+    renderApp();
+    store.setAppControls({
+      isRendered() {
+        return isRendered;
+      },
+      render: renderApp,
+      clear: clearRenderedApp,
+    });
+
+    this.destroy = () => {
+      if (isFF(FF_LSDV_4620_3_ML)) {
+        clearRenderedApp();
+      }
       destroySharedStore();
+      if (isFF(FF_LSDV_4620_3_ML)) {
+        /*
+           It seems that destroying children separately helps GC to collect garbage
+           ...
+         */
+        this.store.selfDestroy();
```

Comment:
This actually makes sense given what we know about MST and its underlying cache of nodes. The process I believe creates a dead set of cache nodes which won't release if attempting to destroy along with the references. So clearing then releasing everything remaining will allow for the Map values to be removed. It is unfortunate that we have to do this manually, but a really great find 🎉 .